### PR TITLE
Create Weet

### PR DESCRIPTION
**Pull request by daniel-vega-86**
## Summary

[Change!] Set post /weets endpoint, validate authentication, create the weet in the database with its corresponding user and add tests

## Known Issues

[Change!] None.

## Trello Card

[Change!] https://trello.com/c/JMMHP971/33-crear-un-weet

### CODE REVIEW

File: app/services/weets.js [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
-    .catch(error => Promise.reject(error));
+const fetchWeet = async user => {
+  try {
+    const { data } = await api.get('/random');
+    if (data.length > 140) {
+      return Promise.reject(
+        errors.defaultError('The content of the weet exceeds 140 characters. Try again.')
+      );
+    }
+    const created = await db.Weet.create({ userId: user.id, content: data });
```
Comment (dbenavidesv) [inline app/services/weets.js]: Services should thought as final branches of a controller logic, so the logic that they execute should be very specific. As I mentioned to you before, the controller function should be the one that controls the flow of logic. This should be in a separate service function and should be called from the controller

File: test/weets.test.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+const db = require('../app/models');
+const { buildUser } = require('./factory/users');
+
+jest.mock('../config/axios');
+
+let user = '';
+
+beforeEach(async () => {
+  await db.User.destroy({ truncate: { cascade: true } });
+  await db.Weet.destroy({ truncate: { cascade: true } });
```
Comment (dbenavidesv) [inline test/weets.test.js]: Why are you doing this truncate?

Comment (daniel-vega-86) [reply test/weets.test.js]: The idea was to clean the tables before each test but I just realized that it was not necessary.

File: app/services/weets.js [hunk 3]
```diff
@@ -1,12 +1,30 @@
...
 
-const fetchWeet = () =>
-  api
-    .get('/random')
-    .then(({ data }) => data)
-    .catch(() => Promise.reject(errors.databaseError('Could not connect with external api')));
+const fetchWeet = async () => {
+  try {
+    const { data } = await api.get('/random');
+    if (!data) throw errors.databaseError('Could not connect with external api');
```
Comment (dbenavidesv) [inline app/services/weets.js]: Here you should throw an error related to the External API

File: app/services/weets.js [hunk 4]
```diff
@@ -1,12 +1,30 @@
...
+    if (data.length > 140) {
+      throw errors.defaultError('The content of the weet exceeds 140 characters. Try again.');
```
Comment (dbenavidesv) [inline app/services/weets.js]: Same here, a more specific type of error

File: test/loginNewUser.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  const { body } = await request(app)
+    .post('/users/sessions')
+    .send({
+      email: loginUser.email,
+      password: loginUser.password
+    });
+  return body;
+};
+
+module.exports = loginNewUser;
```
Comment (dbenavidesv) [inline test/loginNewUser.js]: Nice, but I recommend renaming this file to something more generic and put it in a utils folder, something like this `test/utils/users`

File: test/weets.test.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      .send();
+    expect(response.status).toBe(401);
+    expect(response.body.internal_code).toBe('unauthenticated_error');
+    expect(response.body.message).toBe('Please sign in to access this module.');
+    done();
+  });
+
+  test('Should create a weet', async done => {
+    const body = await loginNewUser(user);
+    await api.get.mockResolvedValue({ data: 'this weet is correct' });
```
Comment (dbenavidesv) [inline test/weets.test.js]: There's no need to await this

File: test/weets.test.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      .set('Authorization', `${body.token}`)
+      .send();
+    expect(response.status).toBe(201);
+    expect(response.body.content).toBe('this weet is correct');
+    done();
+  });
+
+  test('Should fail for a weet with more tha 140 characters', async done => {
+    const body = await loginNewUser(user);
+    await api.get.mockResolvedValue({
```
Comment (dbenavidesv) [inline test/weets.test.js]: same here

**Review by dbenavidesv — APPROVED** [review]

**Pull request merged**